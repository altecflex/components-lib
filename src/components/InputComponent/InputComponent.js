import React from 'react';
import PropTypes from 'prop-types';

/**
 * Input field with label
 * @param {string} label - The label for input
 * @callback callback - function to handle change event
 */

export class InputComponent extends React.Component {
    static propTypes = {
        value: PropTypes.string,
        placeholderTxt: PropTypes.string
    };

    static defaultProps = {
        placeholderTxt: 'Field label'
    };

    constructor(props, context) {
        super(props, context);
        this.state = { active: false, locked: false, error: "" };
    }

    render() {
        const {
            value,
            placeholderTxt
        } = this.props;

        const { active, error, locked } = this.state;

        const fieldClassName = `field ${(locked ? active : active || value) &&
            "active"} ${locked && !active && "locked"}`;

        return (
            <div className={fieldClassName}>
                {active && value }
                <input
                    type="text"
                    value={value}
                    placeholder={placeholderTxt}
                    onFocus={() => !locked && this.setState({ active: true })}
                    onBlur={() => !locked && this.setState({ active: false })}
                />
                <label htmlFor={1} className={error && "error"}>
                    {error || placeholderTxt}
                </label>
            </div>
        )
    }
}
