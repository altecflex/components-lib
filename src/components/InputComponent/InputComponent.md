Example
```jsx
import { InputComponent, ThemeProvider } from "components-lib";

<ThemeProvider>
    <InputComponent placeholderTxt="Name"></InputComponent>
</ThemeProvider>
```
