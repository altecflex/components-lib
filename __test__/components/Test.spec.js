import React from "react";
import { create } from "react-test-renderer";
import { InputComponent } from '../../src/components/InputComponent/InputComponent';

describe("Input component", () => {
    test("Matches the snapshot", () => {
        const input = create(<InputComponent placeholderTxt="Name" />);
        expect(input.toJSON()).toMatchSnapshot();
    });
});