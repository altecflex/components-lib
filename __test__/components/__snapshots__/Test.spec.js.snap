// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Matches the snapshot 1`] = `
<div
  className="field undefined false"
>
  <input
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Name"
    type="text"
  />
  <label
    className=""
    htmlFor={1}
  >
    Name
  </label>
</div>
`;
